src/audio/module_adapter/module/volume/volume.c

```diff
@@ -499,21 +499,15 @@ static int set_volume_ipc4(struct vol_data *cd, uint32_t const channel,
  */
 static inline uint32_t convert_volume_ipc4_to_ipc3(struct comp_dev *dev, uint32_t volume)
 {
-	/* Limit received volume gain to MIN..MAX range before applying it.
-	 * MAX is needed for now for the generic C gain arithmetics to prevent
-	 * multiplication overflow with the 32 bit value. Non-zero MIN option
-	 * can be useful to prevent totally muted small volume gain.
-	 */
-
-	return sat_int24(Q_SHIFT_RND(volume, 31, 23));
+	return sat_int32(Q_SHIFT_RND((int64_t)volume, 31, VOL_QXY_Y));
```

Comment:
this looks strange, if use macro, 31 and 23 both need use macro, if not, we can all use num, if one is macro,, and one is num, looks strange, isn't it?

Reply:
The original code should have used the macro and not hard code the 23. We don't have a macro for IPC4 volume format, so I used just 31.

Reply:
The PR has been tested with extended scope. PASSED in 196 from 202 cases and 51 from 51. Looks great. One case which is failing is 44100hz and I think it is not related to peakvol.

Reply:
we can change and unify based on sof code convention instead of keep as of now, I changed a lot requested by Guennadi in my PR.

Reply:
Y refers in VOL_QXY_Y to the fractional  bits count, in that case 23. It would not make sense to "hardcode" it for use of value 23 as you propose. I like compact macro names due to our line lenght limits so Q_FRACTIONAL_BITS is not as compact.

Changing macro names those are in many locations in code would need another patch or even PR.